src/main/kotlin/org/sert2521/infiniterecharge2020/colorwheelspin/ColorWheelSpinCommands.kt

```diff
@@ -0,0 +1,93 @@
+package org.sert2521.infiniterecharge2020.colorwheelspin
+
+import edu.wpi.first.wpilibj.util.Color
+import java.lang.System.currentTimeMillis
+import kotlin.math.abs
+import kotlinx.coroutines.cancel
+import org.sert2521.sertain.coroutines.doAll
+import org.sert2521.sertain.events.onTick
+import org.sert2521.sertain.subsystems.doTask
+import org.sert2521.sertain.subsystems.use
+
+suspend fun retract(time: Long) = doTask {
+    val colorWheelSpin = use<ColorWheelSpin>()
+    val startTime = currentTimeMillis()
+    action {
+        onTick {
+            colorWheelSpin.extend(-0.2)
+            if (currentTimeMillis() - startTime < time) {
+                this@action.cancel()
+            }
+        }
+    }
+}
+
+suspend fun spinToColor() = doTask {
+    val colorWheelSpin = use<ColorWheelSpin>()
+    action {
+        val colorToSpinTo: Color = Color.kBlue
+        onTick {
+            val sensorColor = getColor(colorWheelSpin.sensor.color)
+            if (sensorColor != colorToSpinTo) {
```

Comment:
Should this be `if (sensorColor == colorToSpinTo)`? Currently I think it's cancelling if the sensor color isn't the one we want to spin to